src/ImageSharp/Processing/Processors/Convolution/MedianRowOperation{TPixel}.cs

```diff
@@ -0,0 +1,164 @@
+        private readonly bool preserveAlpha;
+
+        public MedianRowOperation(Rectangle bounds, Buffer2D<TPixel> targetPixels, Buffer2D<TPixel> sourcePixels, KernelSamplingMap map, int kernelSize, Configuration configuration, bool preserveAlpha)
+        {
+            this.bounds = bounds;
+            this.configuration = configuration;
+            this.targetPixels = targetPixels;
+            this.sourcePixels = sourcePixels;
+            this.map = map;
+            this.kernelSize = kernelSize;
+            this.preserveAlpha = preserveAlpha;
+            int kernelCount = this.kernelSize * this.kernelSize;
+            this.yChannelStart = kernelCount;
+            this.zChannelStart = this.yChannelStart + kernelCount;
+            this.wChannelStart = this.zChannelStart + kernelCount;
+        }
+
+        public void Invoke(int y, Span<Vector4> span)
+        {
+            // Span has kernelSize^2 followed by bound width.
+            int boundsLeft = this.bounds.Left;
+            int boundsWidth = this.bounds.Width;
+            int boundsRight = this.bounds.Right;
+            int kernelCount = this.kernelSize * this.kernelSize;
+            Span<Vector4> kernelBuffer = span.Slice(0, kernelCount);
+            Span<Vector4> channelVectorBuffer = span.Slice(kernelCount, kernelCount);
+            Span<Vector4> targetBuffer = span.Slice(kernelCount << 1, boundsWidth);
+
+            // Stack 4 channels of floats in the space of Vector4's.
+            Span<float> channelBuffer = MemoryMarshal.Cast<Vector4, float>(channelVectorBuffer);
+            var xChannel = channelBuffer.Slice(0, kernelCount);
+            var yChannel = channelBuffer.Slice(this.yChannelStart, kernelCount);
+            var zChannel = channelBuffer.Slice(this.zChannelStart, kernelCount);
+
+            var xOffsets = this.map.GetColumnOffsetSpan();
+            var yOffsets = this.map.GetRowOffsetSpan();
+            var baseXOffsetIndex = 0;
+            var baseYOffsetIndex = (y - this.bounds.Top) * this.kernelSize;
+
+            if (this.preserveAlpha)
+            {
+                for (var x = boundsLeft; x < boundsRight; x++)
+                {
+                    var index = 0;
+                    for (var w = 0; w < this.kernelSize; w++)
+                    {
+                        var j = yOffsets[baseYOffsetIndex + w];
+                        var row = this.sourcePixels.DangerousGetRowSpan(j);
+                        for (var z = 0; z < this.kernelSize; z++)
+                        {
+                            var k = xOffsets[baseXOffsetIndex + z];
+                            var pixel = row[k];
+                            kernelBuffer[index + z] = pixel.ToVector4();
+                        }
+
+                        index += this.kernelSize;
+                    }
+
+                    targetBuffer[x - boundsLeft] = this.FindMedian3(kernelBuffer, xChannel, yChannel, zChannel, kernelCount);
```

Comment:
Looks like we're not actually preserving the alpha here. In other blur operations we preserve the exact per-pixel alpha value of the source. See `ConvolutionProcessor<T>`

Reply:
I'm using the middle **input** pixel for the `W` channel, which boils down to a NO-OP. This preserves the W or alpha value. Added a comment to explain this.